### ISSUE TITLE
KongPlugin resource is created, but its configuration is empty

### ISSUE THREAD

**Issue by tomcastro** [Kong/kubernetes-ingress-controller#294]
NOTE: GitHub issues are reserved for **bug reports only**.
For anything else, please join the conversation
in Kong Nation https://discuss.konghq.com/c/kubernetes.

---

### Summary

Using request transformer plugin, I can create it using a Kubernetes KongPlugin resource, assign it to an Ingress route and it shows up on the Kong Admin API. The problem is the configuration is empty and it doesn't take into account my KongPlugin config.

**Kong Ingress controller version**
0.4.0

**Kong or Kong Enterprise version**
1.1.2

### Kubernetes version

Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.7-gke.10", GitCommit:"8d9b8641e72cf7c96efa61421e87f96387242ba1", GitTreeState:"clean", BuildDate:"2019-04-12T22:59:24Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}

### Environment

GKE Cluster

### What happened
I created the following KongPlugin:
```
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: add-request-transformer
  namespace: dummy
config:
  remove:
    headers:
      - cookie
plugin: request-transformer
```
And assigned it to my Ingress resource:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: dummy
  namespace: dummy
  annotations:
    kubernetes.io/ingress.class: kong
    plugins.konghq.com: oidc-keycloak, add-request-transformer
spec:
  rules:
    - host: <redacted>
      http:
        paths:
          - path: /dummy
            backend:
              serviceName: http-svc
              servicePort: 80
```
If I go to my Kong Admin API, I can see my two plugins (OIDC and Request Transformer). Only OIDC is configured correctly, as the Request Transformer plugin did not take into account the configuration I specified.
```
{"created_at":1558028158,"config":{"remove":{"querystring":[],"headers":[],"body":[]},"replace":{"querystring":[],"headers":[],"body":[]},"http_method":null,"add":{"querystring":[],"headers":[],"body":[]},"append":{"querystring":[],"headers":[],"body":[]},"rename":{"querystring":[],"headers":[],"body":[]}},"id":"cf1051c6-afb5-4f68-8cf1-4dfca695d11d","service":null,"name":"request-transformer","protocols":["http","https"],"enabled":true,"run_on":"first","consumer":null,"route":{"id":"3da8c2a0-60f4-4a06-9135-e5e08339c178"},"tags":null}
```
As you can see, the remove.headers array is empty. I have tried configuring it with no array (config:remove:headers:cookie), with and without quotes, trying to configure different values, setting the rest as empty arrays (as mentioned here #https://github.com/Kong/kubernetes-ingress-controller/issues/102) to no avail. The only thing that fixes it is sending a POST request to the Admin API (or setting through Konga), but this is temporary as it gets replaced eventually by the Ingress Controller.

### Expected behavior
The plugin should be configured using the KongPlugin resource, with an Admin output like the following (notice the config.remove.headers array):
```
{"created_at":1558028158,"config":{"remove":{"querystring":[],"headers":[cookie],"body":[]},"replace":{"querystring":[],"headers":[],"body":[]},"http_method":null,"add":{"querystring":[],"headers":[],"body":[]},"append":{"querystring":[],"headers":[],"body":[]},"rename":{"querystring":[],"headers":[],"body":[]}},"id":"cf1051c6-afb5-4f68-8cf1-4dfca695d11d","service":null,"name":"request-transformer","protocols":["http","https"],"enabled":true,"run_on":"first","consumer":null,"route":{"id":"3da8c2a0-60f4-4a06-9135-e5e08339c178"},"tags":null}**
```

### Steps To Reproduce

1. Set up a Request Transformer KongPlugin resource.
2. Add the annotation to the Ingress resource.

**Comment by gmlp** [comment]
I'm having the same issue. It used to work. @tomcastro have you found any workaround to this issue?
thanks.

**Comment by tomcastro** [comment]
@gmlp Still having the same issue :/ have tried a ton of things, none of them have worked. I have a hunch that Kong Ingress Controller does not like nested YAML files (eg. arrays inside a config file), but I could be wrong. Any help is appreciated :)

**Comment by gmlp** [comment]
@tomcastro, it works with helm chart version 0.9.8. I don't know what is broken in 0.4.0.

**Comment by tomcastro** [comment]
@gmlp I also tried today, mixing and matching different versions, and there's definitely something wrong with version 0.4.0. I'm going to have to downgrade for now.

@hbagdi any way you can help us with this? I suspected the new Kong version had something to do with it (they updated the request transformer plugin in 1.1.2, adding some field validations), but I still had the same problem with 0.4.0. I can also see on the Ingress Controller logs that the correct configuration is getting to the server, but there's probably something that is wiping everything with a list.

**Comment by hbagdi** [comment]
The fix will be available with 0.5.0
